app/controllers/api/v2/courses_controller.rb

```diff
@@ -7,18 +7,22 @@ class CoursesController < API::V2::ApplicationController
 
       before_action :build_recruitment_cycle
       before_action :build_provider
-      before_action :build_course, except: %i[index new create]
+      before_action :build_course, except: %i[index build_new create]
 
       deserializable_resource :course,
                               only: %i[update publish publishable create],
                               class: API::V2::DeserializableCourse
 
-      def new
+      def build_new
         authorize Course
 
         @course = @provider.courses.new
 
-        render jsonapi: @course, include: params[:include]
+        if build_new_course(course_params)
```

Comment:
```suggestion
        build_new_course(course_params)
        if @course.valid?
```

think it reads better
build the thing than ask is it valid

rather then build it and ask is it built...